### PR TITLE
Added tests for SystemInfo.isSandbox

### LINKED ISSUE #536: +(bool) IsSandbox untested

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes #536.

Also made sure to add a test for the case where `Bundle.appStoreReceiptURL` is `nil`.

### CODE REVIEW

**Comment by NachoSoto** [comment]
@taquitos let me know if this is what you had in mind.

File: Purchases/Networking/Backend.swift [hunk 1]
```diff
@@ -384,7 +384,7 @@ class Backend {
         }
 
         if receiptData.count == 0 {
-            if SystemInfo.isSandbox {
+            if self.httpClient.systemInfo.isSandbox {
```
Comment (taquitos) [inline Purchases/Networking/Backend.swift:387]: Can omit `self`

File: Purchases/Misc/SystemInfo.swift [hunk 2]
```diff
@@ -165,7 +171,7 @@ extension SystemInfo {
     }
 
     var isAppExtension: Bool {
-        return Bundle.main.bundlePath.hasSuffix(".appex")
+        return self.bundle.bundlePath.hasSuffix(".appex")
```
Comment (taquitos) [inline Purchases/Misc/SystemInfo.swift:174]: Can omit `self`

File: Purchases/Misc/SystemInfo.swift [hunk 3]
```diff
@@ -47,9 +47,11 @@ class SystemInfo {
     var finishTransactions: Bool
     let platformFlavor: String
     let platformFlavorVersion: String?
+    private let bundle: Bundle
```
Comment (taquitos) [inline Purchases/Misc/SystemInfo.swift:50]: Style: we try to keep the lets/vars organized by:
public
internal
fileprivate
private

and then the static/non-statics grouped together.

File: PurchasesTests/Misc/SystemInfoTests.swift [hunk 4]
```diff
@@ -77,4 +77,28 @@ class SystemInfoTests: XCTestCase {
...
+        expect(SystemInfo.withReceiptResult(.receiptWithData).isSandbox) == false
+    }
+    
+    func testIsNotSandboxIfNoReceiptURL() {
+        expect(SystemInfo.withReceiptResult(.nilURL).isSandbox) == false
+    }
+}
+
+private extension SystemInfo {
+    static func withReceiptResult(_ result: MockBundle.ReceiptURLResult) -> SystemInfo {
```
Comment (taquitos) [inline PurchasesTests/Misc/SystemInfoTests.swift:95]: Style: newline after opening brace for types and newline before closing brace

File: PurchasesTests/Mocks/MockBundle.swift [hunk 5]
```diff
@@ -14,21 +14,30 @@
 import Foundation
 
 class MockBundle: Bundle {
-    enum MockAppStoreReceiptURLResult {
-        case receiptWithData, emptyReceipt, nilURL
+    enum ReceiptURLResult {
```
Comment (taquitos) [inline PurchasesTests/Mocks/MockBundle.swift:17]: 🧡

**Review by taquitos — COMMENTED** [review]
Yes, this is great! Some style nits.

**Comment by NachoSoto** [comment]
Thanks CI for catching that I broke watchOS!

**Comment by taquitos** [comment]
> Thanks CI for catching that I broke watchOS!

best OS

File: PurchasesTests/Purchasing/ReceiptFetcherTests.swift [hunk 6]
```diff
@@ -22,11 +22,16 @@ class ReceiptFetcherTests: XCTestCase {
...
     
     override func setUp() {
         mockBundle = MockBundle()
         mockRequestFetcher = MockRequestFetcher()
-        receiptFetcher = ReceiptFetcher(requestFetcher: mockRequestFetcher, bundle: mockBundle)
+        mockSystemInfo = try! MockSystemInfo(platformFlavor: nil,
+                                             platformFlavorVersion: nil,
+                                             finishTransactions: false,
+                                             bundle: mockBundle)
+        receiptFetcher = ReceiptFetcher(requestFetcher: mockRequestFetcher, systemInfo: mockSystemInfo)
```
Comment (NachoSoto) [inline PurchasesTests/Purchasing/ReceiptFetcherTests.swift:34]: Explicitly passing this here now to fix watchOS

**Comment by NachoSoto** [comment]
Tests should be passing now. Waiting on a decision on explicit `self` to address the remaining comments :)

**Pull request merged**